pkg/catalog/install.go

```diff
@@ -75,64 +64,12 @@ func Install(cfg InstallConfig) error {
 			},
 		}
 	}
-	profileRootdir := filepath.Join(cfg.Directory, cfg.ProfileName)
 	artifacts, err := cfg.ArtifactsMaker.MakeArtifacts(subscription)
 	if err != nil {
 		return fmt.Errorf("failed to generate artifacts: %w", err)
 	}
-	artifactsRootDir := filepath.Join(profileRootdir, "artifacts")
-
-	for _, artifact := range artifacts {
-		artifactDir := filepath.Join(artifactsRootDir, artifact.Name)
-		if err = os.MkdirAll(artifactDir, 0755); err != nil {
-			return fmt.Errorf("failed to create directory")
-		}
-		if artifact.RepoURL != "" {
-			if err := getRepositoryLocalArtifacts(cfg, artifact, artifactDir); err != nil {
-				return fmt.Errorf("failed to get package local artifacts: %w", err)
-			}
-		}
-		for _, obj := range artifact.Objects {
-			filename := filepath.Join(artifactDir, fmt.Sprintf("%s.%s", obj.GetObjectKind().GroupVersionKind().Kind, "yaml"))
-			if err := generateOutput(filename, obj); err != nil {
-				return err
-			}
-		}
-	}
-
-	return generateOutput(filepath.Join(profileRootdir, "profile.yaml"), &subscription)
-}
-
-// getRepositoryLocalArtifacts clones all repository local artifacts so they can be copied over to the flux repository.
-func getRepositoryLocalArtifacts(cfg InstallConfig, a profile.Artifact, artifactDir string) error {
-	u := uuid.NewString()[:6]
-	tmp, err := ioutil.TempDir("", "sparse_clone_git_repo_"+u)
-	if err != nil {
-		return fmt.Errorf("failed to create temp folder: %w", err)
-	}
-	defer func() {
-		if err := os.RemoveAll(tmp); err != nil {
-			fmt.Println("Failed to remove tmp folder: ", tmp)
-		}
-	}()
-	profilePath := a.SparseFolder
-	branch := a.Branch
-	if err := cfg.GitClient.SparseClone(a.RepoURL, branch, tmp, profilePath); err != nil {
-		return fmt.Errorf("failed to sparse clone folder with url: %s; branch: %s; path: %s; with error: %w",
-			a.RepoURL,
-			branch,
-			profilePath,
-			err)
-	}
-	for _, path := range a.PathsToCopy {
-		// nginx/chart/...
-		if strings.Contains(path, string(os.PathSeparator)) {
-			path = filepath.Dir(path)
-		}
-		fullPath := filepath.Join(tmp, profilePath, path)
-		if err := copy.Copy(fullPath, filepath.Join(artifactDir, path)); err != nil {
-			return fmt.Errorf("failed to move folder: %w", err)
-		}
+	if err := cfg.ArtifactsMaker.GenerateArtifactsOutput(artifacts, subscription); err != nil {
```

Comment:
```suggestion
	if err := cfg.ArtifactsMaker.Generate(artifacts, subscription); err != nil {
```

or something like that, to reduce repetetion